File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraScheduler.Drawing;
using DevExpress.XtraScheduler;
using System.Globalization;


namespace HijriScheduler {
    public partial class Form1 : Form {
        HijriCalendarHelper hijriCalHelper;

        public Form1() {
            this.hijriCalHelper = new HijriCalendarHelper();
            InitializeComponent();
        }

        private void schedulerControl1_CustomDrawTimeCell(object sender, DevExpress.XtraScheduler.CustomDrawObjectEventArgs e) {
            SchedulerViewCellBase cell = (SchedulerViewCellBase)e.ObjectInfo;
            bool isOddMonth = hijriCalHelper.IsOddHijriMonths(cell.Interval);
            Color color = isOddMonth ? Color.White : Color.LightSteelBlue;
            cell.Appearance.BackColor = color;
        }
        private void schedulerControl1_VisibleIntervalChanged(object sender, EventArgs e) {
            TimeInterval interval = schedulerControl1.ActiveView.GetVisibleIntervals().Interval;
            hijriCalHelper.FillHijriMonths(interval);
        }
        private void schedulerControl1_SelectionChanged(object sender, EventArgs e) {
            lblInterval.Text = hijriCalHelper.FormatTimeInterval(schedulerControl1.SelectedInterval);
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraScheduler.Drawing
Imports DevExpress.XtraScheduler
Imports System.Globalization


Namespace HijriScheduler
	Partial Public Class Form1
		Inherits Form
		Private hijriCalHelper As HijriCalendarHelper

		Public Sub New()
			Me.hijriCalHelper = New HijriCalendarHelper()
			InitializeComponent()
		End Sub

		Private Sub schedulerControl1_CustomDrawTimeCell(ByVal sender As Object, ByVal e As DevExpress.XtraScheduler.CustomDrawObjectEventArgs) Handles schedulerControl1.CustomDrawTimeCell
			Dim cell As SchedulerViewCellBase = CType(e.ObjectInfo, SchedulerViewCellBase)
			Dim isOddMonth As Boolean = hijriCalHelper.IsOddHijriMonths(cell.Interval)
			Dim color As Color
			If isOddMonth Then
				color = Color.White
			Else
				color = Color.LightSteelBlue
			End If
			cell.Appearance.BackColor = color
		End Sub
		Private Sub schedulerControl1_VisibleIntervalChanged(ByVal sender As Object, ByVal e As EventArgs) Handles schedulerControl1.VisibleIntervalChanged
			Dim interval As TimeInterval = schedulerControl1.ActiveView.GetVisibleIntervals().Interval
			hijriCalHelper.FillHijriMonths(interval)
		End Sub
		Private Sub schedulerControl1_SelectionChanged(ByVal sender As Object, ByVal e As EventArgs) Handles schedulerControl1.SelectionChanged
			lblInterval.Text = hijriCalHelper.FormatTimeInterval(schedulerControl1.SelectedInterval)
		End Sub
	End Class
End Namespace